From e3e0351ddb85e4d07055bbe8646e7438a1f7f81e Mon Sep 17 00:00:00 2001
From: Vratislav Podzimek <v.podzimek@mykolab.com>
Date: Wed, 15 May 2024 09:56:32 +0200
Subject: [PATCH 3/4] Define _XOPEN_SOURCE as 600 on Solaris SPARC

Otherwise it thinks we are compiling for older POSIX with C99
which is a forbidden combination:

```
In file included from /opt/csw/lib/gcc/sparc-sun-solaris2.10/5.5.0/include-fixed/stdio.h:30:0,
                 from crypto/asn1/a_time.c:19:
/opt/csw/lib/gcc/sparc-sun-solaris2.10/5.5.0/include-fixed/sys/feature_tests.h:346:2: error: #error "Compiler or options invalid for pre-UNIX 03 X/Open applications 	and pre-2001 POSIX applications"
 #error "Compiler or options invalid for pre-UNIX 03 X/Open applications \
  ^
```

(cherry picked from commit 42e5fcd12461434ea5d6e04742ea57d4a46ba48d)
---
 crypto/asn1/a_time.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crypto/asn1/a_time.c b/crypto/asn1/a_time.c
index 96ee63d310..8cda70fee0 100644
--- a/crypto/asn1/a_time.c
+++ b/crypto/asn1/a_time.c
@@ -14,6 +14,8 @@
  *      generalTime    GeneralizedTime }
  */
 
+#define _XOPEN_SOURCE 600          /* To get a definition of timezone */
+
 #include <stdio.h>
 #include <time.h>
 #include "crypto/asn1.h"
-- 
2.45.0

